modules/video/src/ecc.cpp

```diff
@@ -309,16 +309,53 @@ static void update_warping_matrix_ECC (Mat& map_matrix, const Mat& update, const
 }
 
 
+/** Function that computes enhanced corelation coefficient from Georgios et.al. 2008
+*   See https://github.com/opencv/opencv/issues/12432
+*/
+double cv::computeECC(InputArray templateImage, InputArray inputImage, InputArray inputMask)
+{
+    Mat src = templateImage.getMat();
+    Mat dst = inputImage.getMat();
+    Mat inputMaskMat = inputMask.getMat();
+
+    CV_Assert(!src.empty());
+    CV_Assert(!dst.empty());
+
+    if( ! (src.type()==dst.type()))
+        CV_Error( Error::StsUnmatchedFormats, "Both input images must have the same data type" );
+
+    if(inputMask.empty())
+        inputMaskMat = Mat::ones(src.rows, src.cols, CV_8U);
```

Comment:
You don't need this.
Keep `inputMask = noArray()` - it is more faster.

Also cache `countNonZero(inputMask)` value (with override for the case of "no mask"):
`int active_pixels = inputMask.empty() ? templateImage.size().area() : countNonZero(inputMask);`

Reply:
That makes sense to me. Thank you once again.

Reply:
Is it possible that python wrapper requires inputMask to be explicitly passed?
```
    (cc, warp_matrix) = cv2.findTransformECC(im1_gray,im2_gray,warp_matrix, warp_mode, criteria)
TypeError: Required argument 'inputMask' (pos 6) not found
```